e2e/cypress/integration/sites-and-applications.spec.js

```diff
@@ -241,67 +233,116 @@ describe('Sites and Applications', () => {
+        .then(($el) => $el[0].checkValidity())
+        .should('be.true')
+
+      cy.findByRole('button', { name: 'Save custom link' }).click()
+    })
+
+    cy.findByRole('link', {
+      name: 'My Custom Link (opens in a new window)',
+    }).should('exist')
+  })
+
+  it('can edit custom links', () => {
     cy.contains('Example Collection')
       .parent()
       .parent()
       .parent()
       .within(() => {
-        // Add a link
-        cy.findByRole('button', { name: '+ Add link' }).click()
-        cy.findByLabelText('URL').clear().type('http://www.example.com')
-        cy.findByRole('option', { name: 'http://www.example.com' }).click()
-        cy.findByLabelText('URL').should('have.value', 'http://www.example.com')
-        cy.findByRole('button', { name: 'Add site' }).click()
+        // Edit a link
+        cy.findAllByRole('button', { name: 'Edit this link' }).first().click()
       })
 
-    cy.findByRole('dialog', { name: 'We don’t recognize that link' }).within(
-      () => {
-        cy.findByLabelText('Label')
-          .then(($el) => $el[0].checkValidity())
-          .should('be.false')
+    cy.findByRole('dialog', { name: 'Edit custom link' }).within(() => {
+      cy.findByRole('button', { name: 'Cancel' }).click()
+    })
 
-        cy.findByLabelText('Label')
-          .type('My Custom Link')
-          .then(($el) => $el[0].checkValidity())
-          .should('be.true')
+    cy.contains('Example Collection')
+      .parent()
+      .parent()
+      .within(() => {
+        // Edit a link
+        cy.findAllByRole('button', { name: 'Edit this link' }).first().click()
+      })
 
-        cy.findByRole('button', { name: 'Save link name' }).click()
-      }
-    )
+    cy.findByRole('dialog', { name: 'Edit custom link' }).within(() => {
+      cy.findByLabelText('Name').should('have.value', 'Webmail')
+      cy.findByLabelText('Name').clear().type('Edited Custom Link')
+      cy.findByLabelText('URL').should('have.value', 'https://google.com')
+      cy.findByLabelText('URL').clear().type('https://example.com')
+      cy.findByRole('button', { name: 'Save custom link' }).click()
+    })
 
     cy.findByRole('link', {
-      name: 'My Custom Link (opens in a new window)',
+      name: 'Edited Custom Link (opens in a new window)',
     }).should('exist')
   })
 
+  it('can delete custom links', () => {
+    cy.contains('Example Collection')
+      .parent()
+      .parent()
+      .within(() => {
+        cy.findAllByRole('listitem').should('have.length', 7)
+
+        // Edit a link
+        cy.findAllByRole('button', { name: 'Edit this link' }).first().click()
+      })
+
+    cy.findByRole('dialog', { name: 'Edit custom link' }).within(() => {
+      cy.findByRole('button', { name: 'Delete' }).click()
+    })
+
+    cy.findByRole('link', {
+      name: 'Edited Custom Link (opens in a new window)',
+    }).should('not.exist')
+    cy.contains('Example Collection')
+      .parent()
+      .parent()
```

Comment:
I think you'll need an additional `.parent()` now to have the cypress test pass

Reply:
thank you!!